draft-ietf-tls-esni.md

```diff
@@ -275,25 +281,30 @@ or if names can be added or removed from the anonymity set during the lifetime
 of a particular ECH configuration.
 
 public_name
-: The non-empty name of the client-facing server, i.e., the entity trusted to
-update the ECH configuration. This is used to correct misconfigured clients, as
-described in {{handle-server-response}}. This value MUST NOT begin or end with
-an ASCII dot and MUST be parsable as a dot-separated sequence of LDH labels, as
-defined in {{!RFC5890}}, Section 2.3.1. Clients MUST ignore any `ECHConfig`
-structure whose `public_name` does not meet these criteria. Note that these
-criteria are incomplete; they incidentally rule out textual representations of
-IPv6 addresses (see {{!RFC3986}}, Section 3.2.2), but do not exclude IPv4
-addresses in standard dotted-decimal or other non-standard notations such as
-octal and hexadecimal (see {{RFC3986}}, Section 7.4). If `public_name` contains
-a literal IPv4 or IPv6 address, the client SHOULD ignore the `ECHConfig` to
-avoid sending a non-compliant "server_name" extension on the ClientHelloOuter
-(see {{!RFC6066}}, Section 3).
+: The non-empty DNS name of the client-facing server, i.e., the entity trusted
+to update the ECH configuration. This is used to correct misconfigured clients,
+as described in {{handle-server-response}}. This value MUST NOT begin or end
+with an ASCII dot and MUST be parsable as a dot-separated sequence of LDH
+labels, as defined in {{!RFC5890}}, Section 2.3.1. Clients MUST ignore any
+`ECHConfig` structure whose `public_name` does not meet these criteria. Note
+that these criteria allow IPv4 addresses in standard dotted-decimal or other
+non-standard notations such as octal and hexadecimal (see {{?RFC3986}}, Section
+7.4). Clients SHOULD ignore the `ECHConfig` if it contains an encoded IPv4
+address. (To determine if a public_name value is an IPv4 address, clients can
+invoke the IPv4 parser algorithm in {{WhatWGURLAddressParser}}. It returns a
+value when the input is an IPv4 address.) Failure to validate that public_name
+is not an IPv4 address may cause connection failures to the client-facing
+server. However, it is not expected to induce security failures. Clients will
+use this name to authenticate the client-facing server and will fail closed if
+the client-facing server cannot produce a certificate bound to the same name.
```

Comment:
"to the same name" might be tricky - if there is a disagreement about whether the name is an IP or DNS name.

How about:

```suggestion
server. However, it is not expected to induce security failures. Clients will
fail closed if the client-facing server cannot produce a certificate bound
to the a reference identity based on the public_name.
```

Reply:
Much better!